src/components/PatientProfilePage/PatientProfilePage.tsx

```diff
@@ -96,39 +141,92 @@ const PatientProfilePage = ({
         status,
         runNumber,
       });
+      if (hospitalId) setSelectedHospital(hospitalId);
+      if (ambulanceId) setSelectedAmbulance(ambulanceId);
+      setTransportConfirmed(status === Status.TRANSPORTED);
     }
-  }, [data]);
+  }, [data, loading, mode]);
+
   useEffect(() => {
     if (mode === 'new' && barcodeValue) {
       setFormFields({ ...formFields, barcodeValue });
     }
-  }, []);
+  }, [mode, barcodeValue]);
 
-  const [addPatient] = useMutation(ADD_PATIENT, {
-    update(cache, { data: { newPatient } }) {
-      cache.writeQuery({
-        query: GET_ALL_PATIENTS,
-        data: { patients: patients.concat([newPatient]) },
-      });
-    },
-  });
-  const [editPatient] = useMutation(EDIT_PATIENT);
+  const handleCloseDialog = () => {
+    setOpenTransportModal(false);
+  };
+
+  const handleCancelTransport = () => {
+    setSelectedHospital({
+      id: '',
+      name: '',
+    });
+    setSelectedAmbulance({
+      id: '',
+      vehicleNumber: 0,
+    });
+    setOpenTransportModal(false);
+    setOpenTransportPage(false);
+  };
+
+  const handleConfirmTransport = () => {
+    setTransportConfirmed(true);
+    setOpenTransportPage(false);
+  };
+
+  const handleHospitalChange = (e: React.ChangeEvent<HTMLElement>) => {
+    setSelectedHospital({
+      id: (e.target as HTMLInputElement).value,
+      name: hospitals.filter(
+        (hospital) => hospital.id === (e.target as HTMLInputElement).value
+      )[0].name,
+    });
+  };
+
+  const handleAmbulanceChange = (e: React.ChangeEvent<HTMLElement>) => {
+    setSelectedAmbulance({
+      id: (e.target as HTMLInputElement).value,
+      vehicleNumber: ambulances.filter(
+        (ambulance) => ambulance.id === (e.target as HTMLInputElement).value
+      )[0].vehicleNumber,
+    });
+  };
+
+  const action = () => (
+    <>
+      <Button
+        style={{ color: '#28B2FF' }}
```

Comment:
Do you want to add this colour to our `Colours` enum in `Constants.ts`? I'm assuming it might be used elsewhere.